trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/rest/TrinoS3ProxyClient.java

```diff
@@ -118,8 +120,11 @@ public void proxyRequest(SigningMetadata signingMetadata, ParsedS3Request reques
             switch (headerName) {
                 case "x-amz-security-token" -> {}  // we add this below
                 case "authorization" -> {} // we will create our own authorization header
-                case "amz-sdk-invocation-id", "amz-sdk-request", "x-amz-decoded-content-length", "content-length", "content-encoding" -> {}   // don't send these
+                case "amz-sdk-invocation-id", "amz-sdk-request", "x-amz-decoded-content-length", "content-length", "transfer-encoding" -> {}   // don't send these
                 case "x-amz-date" -> {} // we will add our own later
+                case "content-encoding" -> remoteRequestHeadersBuilder.add(headerName, headerValues.stream()
```

Comment:
This same parser (or close to it) is in `getRequestContentTypeFromHeaders()` right? Maybe this can be a utility?

Reply:
We need to be more careful with what `content-encoding` we sent to S3, since the PR in its current form will forward all `content-encoding` values, including `aws-chunked` if present.

I'll open a PR to clean this up and separate the forwardable headers vs. the non-forwardable ones - similar to what we do with query parameters already

Reply:
isn't this simpler? `String.join(",", headerValues)` - but you said you will fix this up in another PR?